### PR TITLE
Update to CastXML 0.4.2

### PR DESCRIPTION
**Pull request by josephsnyder**

### CODE REVIEW

**Comment by jcfr** [comment]
Looks like the official release of CastXML is not based of the corresponding tag:

Here is the error reported on AppVeyor:

```
E           AssertionError: assert 'castxml vers...0113-gcbeee87' == 'castxml version 0.4.1'
E             - castxml version 0.4.20210113-gcbeee87
E             + castxml version 0.4.1
tests\test_distribution.py:16: AssertionError

```

**Comment by josephsnyder** [comment]
Yep.  I was hoping to save some time by using a "master" build instead of the release but I'll just have to be patient.

File: CastXMLUrls.cmake [hunk 1]
```diff
@@ -5,14 +5,14 @@
...
 
-set(macosx_binary_url    "https://data.kitware.com/api/v1/file/5f6c9e3d50a41e3d19a8e85c/download")
-set(macosx_binary_sha512 "cf7abb0eeb76ede7a36b6f1ac2545f2dcd00303c817810d118d12aaa1fe058d3348b6555e3ebe7bd47aba713ee02d489f1a33ba58d94d5b6e722fd1cae22647e")
+set(macosx_binary_url    "https://data.kitware.com/api/v1/file/6000a5c62fa25629b9e7f179/download")
+set(macosx_binary_sha512 "79ddc870e363bbbe6001e7a61208c21ec7033a707239247945af518b9d9b5cf310b4827de7c8ba508a9fadeeeb06a19fe0f9b0fa61fc63a62e66087a44defb68")
 
-set(win64_binary_url    "https://data.kitware.com/api/v1/file/5f6c9cd150a41e3d19a8e7b9/download")
-set(win64_binary_sha512 "ed396ebd56301a4dd4c414a02f8c41293d1ed60df0802ccf4eb1f6db63bdf7d2ddea0eb66cc5ce1e1fd244e29d446072eae9fbe9f13159fb0213772a796a10eb")
+set(win64_binary_url    "https://data.kitware.com/api/v1/file/6000b3222fa25629b9e82a50/download")
+set(win64_binary_sha512 "c3c5a0c0a499c3128986a237bb8ea3d2f756aad68ebad69619f3a08014819b84e4bdad3fc179937ade941602b61feab3eba35ba804b9a5083362f6148c971f91")]
```
Comment (jcfr) [inline CastXMLUrls.cmake]:
```suggestion
set(win64_binary_sha512 "c3c5a0c0a499c3128986a237bb8ea3d2f756aad68ebad69619f3a08014819b84e4bdad3fc179937ade941602b61feab3eba35ba804b9a5083362f6148c971f91")
```

**Comment by jcfr** [comment]
I will have a look at the Python 3.x windows error

**Comment by jcfr** [comment]
> re: appveyor failure

The file `enable-worker-remote-access.ps1` is made available by https://github.com/scikit-build/scikit-ci-addons here: https://github.com/CastXML/CastXML-python-distributions/blob/37c6977f009c34b9b891f2910cf5207c11d90f30/appveyor.yml#L45

Fixing `scikit-ci-addons` and making a new release of that project would address this, since this a project I maintain it should not be be an issue.

**Comment by jcfr** [comment]
> Fixing `scikit-ci-addons` and making a new release of that project would address this, since this a project I maintain it should not be be an issue.

Update: It seems that `scikit-ci` is also having issue .. and this is a project we maintain also. See https://github.com/scikit-build/scikit-ci

**Comment by jcfr** [comment]
> re: travis

I will another look later and report back

**Comment by jcfr** [comment]
>  re: circleci

I just enabled this settings so that you see build results for manylinux:

image

**Comment by josephsnyder** [comment]
@jcfr, Any thoughts about the Travis builds?

File: .travis.yml [hunk 2]
```diff
@@ -9,14 +9,18 @@ matrix:
   include:
 
     - os: osx
+      osx_image: xcode12
       language: generic
       env:
         - PYTHON_VERSION=3.7.8
+        - MACOSX_DEPLOYMENT_TARGET=10.15.2
```
Comment (jcfr) [inline .travis.yml]:
```suggestion
```

File: .travis.yml [hunk 3]
```diff
@@ -9,14 +9,18 @@ matrix:
...
 
     - os: osx
+      osx_image: xcode12
       language: generic
       env:
         - PYTHON_VERSION=2.7.15
+        - MACOSX_DEPLOYMENT_TARGET=10.15.2
```
Comment (jcfr) [inline .travis.yml]:
```suggestion
```

File: .travis.yml [hunk 4]
```diff
@@ -49,7 +55,7 @@ before_install:
   - ci_addons --install ../addons
 
 install:
-  - ci install
+  - travis_wait 15 ci install
```
Comment (jcfr) [inline .travis.yml]: Add a comment explaining why this is needed

File: scikit-ci.yml [hunk 5]
```diff
@@ -17,8 +17,9 @@ before_install:
   travis:
     osx:
       environment:
+        CPPFLAGS: "-stdlib=libstdc++"
         PATH: $<HOME>/.pyenv/versions/$<PYTHON_VERSION>/bin:$<PATH>
-        SETUP_BDIST_WHEEL_ARGS: --plat-name macosx-10.6-x86_64
+        SETUP_BDIST_WHEEL_ARGS: --plat-name macosx-10.15.2-x86_64
```
Comment (jcfr) [inline scikit-ci.yml]:
```suggestion
        SETUP_BDIST_WHEEL_ARGS: --plat-name macosx-10.13-x86_64
```

**Review by jcfr — CHANGES_REQUESTED** [review]
In addition, I suggest we update the test trying to install the wheel to use `--no-index`. See https://github.com/CastXML/CastXML-python-distributions/blob/2b0f0cd74975a30a21feae8bd6ca0201781a406f/tests/test_distribution.py#L27

File: .travis.yml (hunk 4 above)
Comment (josephsnyder) [reply .travis.yml]: It was originally "needed" during one of the interim runs when it seemed to time out there.  I've removed it in the current pull request.

**Comment by josephsnyder** [comment]
I don't get why the `10.15.2` => `10.13` was what it needed, but @jcfr, you solved it!

I'll squash the commits and force push one more time.

File: scikit-ci.yml [hunk 6]
```diff
@@ -17,8 +17,9 @@ before_install:
   travis:
     osx:
       environment:
+        CPPFLAGS: "-stdlib=libstdc++"
```
Comment (jcfr) [inline scikit-ci.yml:21]:
```suggestion
        # See https://scikit-build.readthedocs.io/en/latest/generators.html#libstdc-vs-libc
        CPPFLAGS: "-stdlib=libstdc++"
```

File: scikit-ci.yml [hunk 7]
```diff
@@ -17,8 +17,9 @@ before_install:
...
         PATH: $<HOME>/.pyenv/versions/$<PYTHON_VERSION>/bin:$<PATH>
-        SETUP_BDIST_WHEEL_ARGS: --plat-name macosx-10.6-x86_64
+        SETUP_BDIST_WHEEL_ARGS: --plat-name macosx-10.13-x86_64
```
Comment (jcfr) [inline scikit-ci.yml:23]: I couldn't find a website listing what could be the minimum target deployment for a given based SDK. Considered that `xcode12` Travis image is running `macOS 10.15.7` and bundles  `Xcode 12.0.1`    (see [here](https://docs.travis-ci.com/user/reference/osx#macos-version), targeting `10.13` seems "reasonable". 

If we get complains from users, we can re-assess to build wheel compatible with older system.

Also worth noting that macOS `10.13` is old enough to ensure the wheel will be installable in most of the current system:

image

image

**Review by jcfr — COMMENTED** [review]
Thanks for working on this :pray:

**Comment by jcfr** [comment]
After the CI completes, let's squash and integrate.

**Comment by josephsnyder** [comment]
I will do that.  Thanks again, JC

**Comment by jcfr** [comment]
> I will do that. Thanks again, JC

Thanks. When squashing, could you add the following at the end of the commit message: `Co-authored-by: Jean-Christophe Fillion-Robin <jchris.fillionr@kitware.com>`. For more details about this convention, see https://github.blog/2018-01-29-commit-together-with-co-authors/

**Comment by josephsnyder** [comment]
Done

**Pull request merged**